src/components/api/RoiList.tsx

```diff
@@ -1,9 +1,15 @@
-import { useRois } from '../../hooks';
+import { useRoiState, useRois } from '../../hooks';
 
 import { RoiBox } from './RoiBox';
 
 export function RoiList() {
-  const rois = useRois();
+  const rois = useRois().slice();
+  const { selectedRoi } = useRoiState();
+  if (selectedRoi) {
+    const index = rois.findIndex((roi) => roi.id === selectedRoi);
+    const roi = rois.splice(index, 1)[0];
```

Comment:
Would add an assertion here about the existence of the selected roi
Shouldn't happen to run into this but I you do you'd have an error much easier to debug.

Reply:
Yes
In general f I don't expect something to happen, I throw. It's a good way of making sure bugs are caught early.